main.go

```diff
@@ -35,6 +35,7 @@ import (
 	"github.com/kubernetes-incubator/external-dns/controller"
 	"github.com/kubernetes-incubator/external-dns/pkg/apis/externaldns"
 	"github.com/kubernetes-incubator/external-dns/pkg/apis/externaldns/validation"
+	"github.com/kubernetes-incubator/external-dns/pkg/util/domains"
```

Comment:
If the domain filter only to be used by providers then yes, it should be either in provider package directory, or it's child subdirectory. 
If we find that `provider` exposes too many types/functions we can think of refactoring it later on